src/Sylius/Component/Core/Taxation/Applicator/OrderItemsTaxesApplicator.php

```diff
@@ -69,14 +69,20 @@ public function apply(OrderInterface $order, ZoneInterface $zone): void
         $itemSplitTaxes = $this->proportionalIntegerDistributor->distribute($itemTaxWholeAmounts, $itemTotalTaxWholeAmount);
 
         foreach ($order->getItems() as $index => $item) {
+            $itemSplitTaxesIndex = $index;
+
+            if (!array_key_exists($itemSplitTaxesIndex, $itemSplitTaxes)) {
```

Comment:
There is a problem that index is reaching outside of the `$itemSplitTaxes` keys.
In behat scenario we've got case that `OrderItemUnit` has index `2` when `OrderItem` itself has only one unit, it causes key to not be found as you can see below, `$itemSplitTaxes` here has only one key that is equal zero.
![image](https://github.com/Sylius/Sylius/assets/53942444/0d08c66d-6cdc-4916-aed1-2128cdda8b20)
![image](https://github.com/Sylius/Sylius/assets/53942444/050c3ddd-134e-4fee-be9f-d7042946dbd3)

Sadly, I could not reproduce the problem by myself, it seems to only occur in behat, also this if's seem to be safe, as we only check case when it would throw an error, but I believe it could be solved better 🤷🏽‍♂️